pkg/reconciler/service/service_test.go

```diff
@@ -1403,8 +1424,8 @@ func TestNew(t *testing.T) {
 	}
 }
 
-func config(name, namespace string, so ServiceOption, co ...ConfigOption) *v1alpha1.Configuration {
-	s := Service(name, namespace, so)
+func config(name, namespace string, sos []ServiceOption, co ...ConfigOption) *v1alpha1.Configuration {
+	s := Service(name, namespace, sos...)
 	s.SetDefaults(v1.WithUpgradeViaDefaulting(context.Background()))
```

Comment:
looks like we apply defaulting to the service in `config` and `route`. I wonder if it is necessary to pass the defaulting `ServiceOption` through. or if we do, we should probably call it with `v1.WithUpgradeViaDefaulting(context.Background()` and then remove the line above?